### PR TITLE
refactor: optimize dockerfile for AsyncAPI generator setup

### PR DESCRIPTION
**Pull request by Light13008**
**Description**
-I tried to solve this [issue](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKm)

### CODE REVIEW

**Review by derberg — COMMENTED** [review]
please update your PR to address only the issue mentioned in sonar issue that you linked

File: apps/generator/Dockerfile [hunk 1]
```diff
@@ -1,21 +1,19 @@
-ARG ASYNCAPI_GENERATOR_VERSION=1.10.9
-
 FROM node:18-alpine
 
+ARG ASYNCAPI_GENERATOR_VERSION="1.10.9"  # Include ARG in the build stage and surround it with double quotes
```
Comment (DhairyaMajmudar) [inline apps/generator/Dockerfile:3]: Pls. remove these unwanted comments

File: apps/generator/Dockerfile [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 # Since 0.14.0 release of html-template chromium is needed for pdf generation.
 # More custom packages for specific template should not be added to this dockerfile. Instead, we should come up with some extensibility solution.
-RUN apk --update add git chromium && \
-    rm -rf /var/lib/apt/lists/* && \
-    rm /var/cache/apk/*
-
-# Installing latest released npm package
-RUN npm install -g @asyncapi/generator@$ASYNCAPI_GENERATOR_VERSION
+RUN apk --update add --no-cache git chromium && \
+    npm install -g @asyncapi/generator@"$ASYNCAPI_GENERATOR_VERSION" && \  # Surround variable with double quotes
```
Comment (DhairyaMajmudar) [inline apps/generator/Dockerfile:15]: From here also, pls. remove extra commnets

**Comment by derberg** [comment]
duplicate of https://github.com/asyncapi/generator/pull/1281/files

**Pull request closed without merging**